drivers/SmartThings/zigbee-button/src/zigbee-multi-button/robb/init.lua

```diff
@@ -0,0 +1,183 @@
+      return true
+    end
+  end
+  return false
+end
+
+-- Map left column buttons to endpoints
+local EP_BUTTON_ON_COMPONENT_MAP = {
+  [0x01] = "button1",
+  [0x02] = "button3",
+  [0x03] = "button5",
+  [0x04] = "button7"
+}
+
+-- Map right column buttons to endpoints
+local EP_BUTTON_OFF_COMPONENT_MAP = {
+  [0x01] = "button2",
+  [0x02] = "button4",
+  [0x03] = "button6",
+  [0x04] = "button8"
+}
+
+local build_button_handler = function(driver, device, zb_rx)
+    local button_name
+    local additional_fields = { state_change = true }
+
+    -- Fetch correct button name
+    if zb_rx.body.zcl_header.cmd.value == 0x01 then
+      button_name = EP_BUTTON_ON_COMPONENT_MAP[zb_rx.address_header.src_endpoint.value]
+    else
+      button_name = EP_BUTTON_OFF_COMPONENT_MAP[zb_rx.address_header.src_endpoint.value]
+    end
+
+    local event = capabilities.button.button.pushed(additional_fields)
+    local comp = device.profile.components[button_name]
+    -- Emit events
+    if comp ~= nil then
+      device:emit_component_event(comp, event)
+      if button_name ~= "main" then
+        device:emit_event(event)
+      end
+    end
+end
+
+local function hold_handler(driver, device, zb_rx)
+  local button_name
+  local pressed_type
+  local additional_fields = { state_change = true }
+
+  -- Handle MoveStepMode.UP
+  if zcl_clusters.Level.types.MoveStepMode.UP == zb_rx.body.zcl_body.move_mode.value then
+    button_name = EP_BUTTON_ON_COMPONENT_MAP[zb_rx.address_header.src_endpoint.value]
+    pressed_type = "up_hold"
+  -- Handle MoveStepMode.DOWN
+  else
+    button_name = EP_BUTTON_OFF_COMPONENT_MAP[zb_rx.address_header.src_endpoint.value]
+    pressed_type = "down_hold"
+  end
+
+  local event = capabilities.button.button[pressed_type](additional_fields)
+  local comp = device.profile.components[button_name]
+  -- Emit events
+  if comp ~= nil then
+    device:emit_component_event(comp, event)
+    device:emit_event(event)
+  end
+end
+
+local do_configuration = function(driver, device)
+  local has8Btns = device:get_model() == "ROB_200-007-0"
+  -- Get the right number of endpoints
+  local endpoints = has8Btns and SWITCH8_NUM_ENDPOINT or SWITCH4_NUM_ENDPOINT
+
+  device:send(PowerConfiguration.attributes.BatteryVoltage:configure_reporting(device, 30, 21600, 1))
+  device:send(device_management.build_bind_request(device, PowerConfiguration.ID, driver.environment_info.hub_zigbee_eui))
+
+  for endpoint = 1, endpoints do
+    device:send(device_management.build_bind_request(device, Level.ID, driver.environment_info.hub_zigbee_eui, endpoint))
+    device:send(device_management.build_bind_request(device, OnOff.ID, driver.environment_info.hub_zigbee_eui, endpoint))
+  end
+
+  device:send(OnOff.attributes.OnOff:configure_reporting(device, 0, 600, 1))
+
+  device:send(Basic.attributes.DeviceEnabled:write(device, true))
+  if not driver.datastore[SWITCH_GROUP_CONFIGURE] then
```

Comment:
We prefer you use `device:set_field` and `device:get_field` here rather than access the datastore directly.

You may also want to specify `{persist = true}` if you'd like those changes to be durable.

Reply:
solved with 714d055